src/status_im/contexts/settings/language_and_currency/currency/view.cljs

```diff
@@ -0,0 +1,78 @@
+(ns status-im.contexts.settings.language-and-currency.currency.view
+  (:require [quo.core :as quo]
+            [react-native.core :as rn]
+            [status-im.common.resources :as resources]
+            [status-im.contexts.settings.language-and-currency.currency.style :as style]
+            [status-im.contexts.settings.language-and-currency.currency.utils :as utils]
+            [status-im.contexts.settings.language-and-currency.data-store :as data-store]
+            [utils.debounce :as debounce]
+            [utils.i18n :as i18n]
+            [utils.navigation :as navigation]
+            [utils.re-frame :as rf]))
+
+(defn- get-item-layout
+  [_ index]
+  #js {:length 100 :offset (* 100 index) :index index})
+
+(defn- settings-category-view
+  [{:keys [title data]} _ _ {:keys [selected-currency on-currency-press]}]
+  [rn/delay-render
+   [quo/category
+    {:label     title
+     :data      (mapv (fn [currency]
+                        (utils/make-currency-item
+                         {:currency          currency
+                          :selected-currency selected-currency
+                          :on-change         on-currency-press}))
+                      data)
+     :blur?     true
+     :list-type :settings}]])
+
+(defn view
+  []
+  (let [[search-text set-search-text] (rn/use-state "")
+        on-currency-press             (rn/use-callback #(rf/dispatch [:profile.settings/update-currency
+                                                                      %]))
+        selected-currency             (rf/sub [:profile/currency])
+        currencies                    (rf/sub [:currencies/categorized search-text])
+        on-change-text                (rn/use-callback
+                                       (debounce/debounce
+                                        #(set-search-text %)
+                                        300))
+        formatted-data                (rn/use-memo
+                                       #(data-store/get-formatted-currency-data currencies)
+                                       [search-text])]
```

Comment:
I see why and how `search-text` can be used for caching here, but technically, shouldn't we cache based on `currencies` instead because that's the input to the pure function?